.github/workflows/pull-request.yml

```diff
@@ -27,16 +27,34 @@ jobs:
       # ensuring that PRs can use the cache from main.
       #
       # See https://github.com/actions/setup-go/issues/316 for more info.
-      - uses: actions/setup-go@v4
+      - name: Install go
+        uses: actions/setup-go@v4
         with:
           go-version-file: 'go.mod'
-          cache: ${{ github.ref == 'refs/heads/main' }} # only update the cache in main.          
+          cache: ${{ github.ref == 'refs/heads/main' }} # only update the cache in main.
 
-      - name: Run build
+      - name: Run go build
         run: go build ./...
-  
+
       # - name: Run lint
       #   run: make lint
 
-      - name: Run the unit test
-        run: make test
\ No newline at end of file
+      - name: Run go unittest
+        run: make test
+
+      - name: Install helm
+        uses: azure/setup-helm@v4.3.0
+        with:
+          version: v3.17.3
+
+      - name: Install helm-unittest
+        run: helm plugin install https://github.com/helm-unittest/helm-unittest.git
+
+      - name: Run helm-unittest
+        working-directory: charts/s2s-proxy
+        run: helm unittest .
+
+      - name: Check Helm example
+        run: |
+          make helm-example
+          .github/workflows/scripts/check-diff.sh charts/s2s-proxy/example.yaml
```

Comment:
what is this for? just to show diff on the output?

Reply:
It's to make sure the `example.yaml` is up to date. 

If we make changes to the Helm chart, we should run `helm-example` to update `example.yaml` and commit it.

Let me add some useful failure message there.